packages/mui-material-next/src/FormHelperText/FormHelperText.tsx

```diff
@@ -0,0 +1,235 @@
+import useFormControl from '../FormControl/useFormControl';
+import formHelperTextClasses, { getFormHelperTextUtilityClasses } from './formHelperTextClasses';
+import {
+  FormHelperTextOwnerState,
+  FormHelperTextProps,
+  FormHelperTextTypeMap,
+} from './FormHelperText.types';
+
+const useUtilityClasses = (ownerState: FormHelperTextOwnerState) => {
+  const { classes, contained, size, disabled, error, filled, focused, required } = ownerState;
+  const slots = {
+    root: [
+      'root',
+      disabled && 'disabled',
+      error && 'error',
+      size && `size${capitalize(size)}`,
+      contained && 'contained',
+      focused && 'focused',
+      filled && 'filled',
+      required && 'required',
+    ],
+  };
+
+  return composeClasses(slots, getFormHelperTextUtilityClasses, classes);
+};
+
+const FormHelperTextRoot = styled('p', {
+  name: 'MuiFormHelperText',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      ownerState.size && styles[`size${capitalize(ownerState.size)}`],
+      ownerState.contained && styles.contained,
+      ownerState.filled && styles.filled,
+    ];
+  },
+})<{ ownerState: FormHelperTextOwnerState }>(({ theme, ownerState }) => {
+  const { vars: tokens } = theme;
+
+  const pxFontSize = theme.sys.typescale.body.small.size;
+  const lineHeight = `calc(${tokens.sys.typescale.body.small.lineHeight} / ${pxFontSize})`;
+
+  return {
+    '--md-comp-form-helper-text-color': tokens.sys.color.onSurfaceVariant,
+    '--md-comp-form-helper-text-font-family': tokens.sys.typescale.body.small.family,
+    '--md-comp-form-helper-text-font-size': theme.typography.pxToRem(pxFontSize), // the pxToRem should be moved to typescale in the future,
+    '--md-comp-form-helper-text-font-weight': tokens.sys.typescale.body.small.weight,
+    '--md-comp-form-helper-text-line-height': lineHeight,
+    '--md-comp-form-helper-text-disabled-color': tokens.sys.color.onSurface,
+    '--md-comp-form-helper-text-disabled-opacity': 0.38,
+    '--md-comp-form-helper-text-error-color': tokens.sys.color.error,
+    color: 'var(--md-comp-form-helper-text-color)',
+    fontFamily: 'var(--md-comp-form-helper-text-font-family)',
+    fontSize: 'var(--md-comp-form-helper-text-font-size)',
+    lineHeight: `var(--md-comp-form-helper-text-line-height)`,
+    textAlign: 'left',
+    marginTop: 3,
+    marginRight: 0,
+    marginBottom: 0,
+    marginLeft: 0,
+    [`&.${formHelperTextClasses.disabled}`]: {
+      color:
+        'color-mix(in srgb, var(--md-comp-form-helper-text-disabled-color) calc(- var(--md-comp-form-helper-text-disabled-opacity) * 100%), transparent)',
+    },
+    [`&.${formHelperTextClasses.error}`]: {
+      color: 'var(--md-comp-form-helper-text-error-color)',
+    },
+    ...(ownerState.size === 'small' && {
+      marginTop: 4,
+    }),
+    ...(ownerState.contained && {
+      marginLeft: 14,
+      marginRight: 14,
```

Comment:
14 is an off number to end up with considering the spacing, is this based on the spec?

Reply:
The Figma file uses `16px` on both sides:

<img width="400" alt="Screenshot 2023-11-21 at 10 42 52" src="https://github.com/mui/material-ui/assets/37222944/546276df-5639-4b30-aa14-7a6abad309de">


However, if we change this, we must ensure that all labels are aligned:

<img width="400" alt="Screenshot 2023-11-21 at 10 45 04" src="https://github.com/mui/material-ui/assets/37222944/4ca2fc46-3ca7-428e-8103-ee6407d7ebfe">

Reply:
👍 I've realigned everything https://github.com/mui/material-ui/pull/39503/commits/5f9105dd641be2570d6964a5aa07e1c091bdd7fa

(this is actually off in v5 as well, the MD2 spec also specifies 16px padding)